File: P4/civitasruby/lib/civitas/sorpresa_casa_hotel.rb
#encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

# Salva

module Civitas
  class Sorpresa_casa_hotel < Sorpresa # sorpresa de pagar por tantas casas y hoteles se tengan
    def initialize(valor, texto)
      super(texto)
      @valor = valor
    end
    
    def aplicar_a_jugador(actual,todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        todos.at(actual).modificar_saldo(@valor*todos.at(actual).cantidad_casas_hoteles)
      end
    end
    
  end
end

File: PruebaA2/Civitas/src/civitas/SorpresaCasaHotel.java
package civitas;
import java.util.ArrayList;

/**
 *
 * @author abel
 */
public class SorpresaCasaHotel extends Sorpresa {
    
    private int valor;
    
    SorpresaCasaHotel(int valor, String texto){
        this.texto = texto;
        this.valor = valor;  
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor * todos.get(actual).cantidadCasasHoteles());
        }
    }
}
